src/datastructure/RailwayNetwork.cpp

```diff
@@ -653,3 +657,749 @@ cda_rail::Network::Network(const std::filesystem::path &p) {
+    }
+
+    // Call function corresponding to separation type
+    if (separation_type == cda_rail::SeparationType::UNIFORM) {
+        return uniform_separate_edge(edge_index);
+    } else if (separation_type == cda_rail::SeparationType::CHEBYCHEV) {
+        return chebychev_separate_edge(edge_index);
+    } else {
+        throw std::invalid_argument("Separation type does not exist.");
+    }
+}
+
+std::pair<std::vector<int>, std::vector<int>> cda_rail::Network::uniform_separate_edge(int edge_index) {
+    /**
+     * Separates an edge (and possibly its reverse edge) uniformly according to the minimal block length.
+     */
+
+    // Check if graph is consistent for this function
+    if (!is_consistent_for_transformation()) {
+        throw std::invalid_argument("Graph is not consistent");
+    }
+
+    // Check if edge exists
+    if (!has_edge(edge_index)) {
+        throw std::invalid_argument("Edge does not exist");
+    }
+
+    // If the edge is not breakable throw an exception
+    if (!get_edge(edge_index).breakable) {
+        throw std::invalid_argument("Edge is not breakable");
+    }
+
+    // If the edge length is smaller than twice the minimum block length, nothing to separate
+    if (get_edge(edge_index).length < 2 * get_edge(edge_index).min_block_length) {
+        return std::make_pair(std::vector<int>(), std::vector<int>());
+    }
+
+    // Get edge to separate
+    const auto& edge = get_edge(edge_index);
+    // Get number of new vertices
+    double number_of_blocks = std::floor(edge.length / edge.min_block_length) ;
+
+    // Calculate distances
+    std::vector<double> distances_from_source;
+    for (int i = 1; i < number_of_blocks; ++i) {
+        distances_from_source.emplace_back(i * (edge.length / number_of_blocks));
+    }
+
+    return separate_edge_at(edge_index, distances_from_source);
+}
+
+std::pair<std::vector<int>, std::vector<int>> cda_rail::Network::chebychev_separate_edge(int edge_index) {
+    // TODO: Not implemented yet
+    throw std::runtime_error("Not implemented yet");
+}
+
+std::vector<int> cda_rail::Network::breakable_edges() const {
+    /**
+     * Returns indices of all brekable edges.
+     */
+    // Check if graph is consistent for this function
+    if (!is_consistent_for_transformation()) {
+        throw std::invalid_argument("Graph is not consistent");
+    }
+
+    std::vector<int> ret_val;
+    for (int i = 0; i < number_of_edges(); ++i) {
+        if (get_edge(i).breakable) {
+            ret_val.emplace_back(i);
+        }
+    }
+    return ret_val;
+}
+
+std::vector<int> cda_rail::Network::relevant_breakable_edges() const {
+    /**
+     * Returns inidices of all breakable edges, but only once per direction.
+     */
+    // Check if graph is consistent for this function
+    if (!is_consistent_for_transformation()) {
+        throw std::invalid_argument("Graph is not consistent");
+    }
+
+    std::vector<int> ret_val;
```

Comment:
Problematic here is that we do not copy, because edges contains object of type Edge, whereas in ret_val we only store indices and do not copy any element.